java/arcs/core/storage/handle/CollectionImpl.kt

```diff
@@ -44,33 +44,47 @@ class CollectionImpl<T : Referencable>(
      *
      * It will be passed to the storage proxy in an add operation, with an incremented version
      * for this Handle in the version map.
+     *
+     * If this returns false, your change was not applied due to stale state or an invalid operation
+     * (such as removing a non-existent entity from a collection). You should reread the latest
+     * value of the handle and see if a retry is warranted.
      */
-    suspend fun store(entity: T) {
+    suspend fun store(entity: T): Boolean {
         log.debug { "Storing: $entity" }
         versionMap.increment()
-        storageProxy.applyOp(CrdtSet.Operation.Add(name, versionMap, entity))
+        return storageProxy.applyOp(CrdtSet.Operation.Add(name, versionMap, entity))
     }
 
     /**
      * Clear all items from the collection.
      *
      * This currently works by iterating over all items in the storage proxy view of the
      * collection, and sending a Remove command for each one.
+     *
+     * If this returns false, your change was not applied due to stale state or an invalid operation
```

Comment:
Is it worth noting in the comment that the change may have been partially applied in this case?

Reply:
Yes good idea, done.